addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxPlayerHandler.java

```diff
@@ -0,0 +1,432 @@
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.IncreaseDecreaseType;
+import org.eclipse.smarthome.core.library.types.NextPreviousType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.library.types.PlayPauseType;
+import org.eclipse.smarthome.core.library.types.RawType;
+import org.eclipse.smarthome.core.library.types.RewindFastforwardType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.types.UpDownType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.squeezebox.config.SqueezeBoxPlayerConfig;
+import org.openhab.binding.squeezebox.internal.utils.HttpUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SqueezeBoxPlayerHandler} is responsible for handling states, which
+ * are sent to/from channels.
+ * 
+ * @author Dan Cunningham - Initial contribution
+ */
+public class SqueezeBoxPlayerHandler extends BaseThingHandler implements
+		SqueezeBoxPlayerEventListener {
+
+	private Logger logger = LoggerFactory
+			.getLogger(SqueezeBoxPlayerHandler.class);
+
+	public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections
+			.singleton(SQUEEZEBOXPLAYER_THING_TYPE);
+
+	/**
+	 * We need to remember some states to change offsets in volume, time index,
+	 * etc..
+	 */
+	protected Map<String, State> stateMap = Collections
+			.synchronizedMap(new HashMap<String, State>());
+
+	/**
+	 * Keeps current track time
+	 */
+	ScheduledFuture<?> timeCounterJob;
+
+	/**
+	 * Local reference to our bridge
+	 */
+	private SqueezeBoxServerHandler _squeezeBoxServerHandler;
+
+	/**
+	 * Our mac address, needed everywhere
+	 */
+	private String mac;
+
+	/**
+	 * Before we mute or recieve a mute event, store our current volume
+	 */
+	private int unmuteVolume = 0;
+
+	/**
+	 * The server sends us the current time on play/pause/stop events, we
+	 * increment it locally from there on
+	 */
+	private int currentTime = 0;
+
+	/**
+	 * Our we playing something right now or not, need to keep current track
+	 * time
+	 */
+	private boolean playing;
+
+	/**
+	 * Creates SqueezeBox Player Handler
+	 * 
+	 * @param thing
+	 */
+	public SqueezeBoxPlayerHandler(Thing thing) {
+		super(thing);
+	}
+
+	@Override
+	public void initialize() {
+		mac = getThing().getConfiguration().as(SqueezeBoxPlayerConfig.class).mac;
+		timeCounter();
+		if (squeezeBoxServerHandler() != null) {
+			updateStatus(
```

Comment:
Yes, that will actually remove a lot of code for me, I must have been looking at some early bindings where these  calls did not exist yet.